src/games/strategy/triplea/ai/proAI/ProNonCombatMoveAI.java

```diff
@@ -1905,6 +1905,17 @@ public int compare(final ProTerritory t1, final ProTerritory t2) {
               continue;
             }
 
+            // Check if territory is safe after all current moves
+            final List<Unit> defendingUnits = moveMap.get(t).getAllDefenders();
+            moveMap.get(t).setBattleResult(
```

Comment:
How do you know the battle result is not already populated? The rest of the class seems to take care to not re-calc results, but here we always set it. If we always set it here, can we then remove some of this lazy evaluation in other places?

For example, this code block was also found on at least lines:  656 + 710 + 757 + 802
```
          if (moveMap.get(t).getBattleResult() == null) {
            moveMap.get(t).setBattleResult(
                calc.estimateDefendBattleResults(player, t, moveMap.get(t).getMaxEnemyUnits(),
                    defendingUnits, moveMap.get(t).getMaxEnemyBombardUnits()));
          }
```